src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -371,7 +371,11 @@ public void delete() {
   }
 
   private void checkNotDeleted() {
-    checkState(!deleted.get(), "FirebaseApp was deleted %s", this);
+    // Wrap the name argument in an array to ensure the invocation gets bound to the commonly
```

Comment:
This is fine by me, but it's a very round-about way to solve a problem in a method that does nothing but calling `new IllegalStateException`.

I would just call it myself here.